embulk-jruby-extension/src/main/java/org/jruby/util/RubyDateParser.java

```
@@ -0,0 +1,605 @@
+                {
+                    int monIndex = matchAtPatterns(text, pos, monNames);
+                    if (monIndex >= 0) {
+                        values.mon = monIndex % 12 + 1;
+                        pos += monNames[monIndex].length();
+                    } else {
+                        values.fail();
+                    }
+                    break;
+                }
+                case FORMAT_CENTURY: // %C - year / 100 (round down.  20 in 2009)
+                {
+                    long c;
+                    if (matchAtNumPatterns(nextToken(compiledPattern, i))) {
+                        c = readDigits(2);
+                    } else {
+                        c = readDigitsMax();
+                    }
+                    values._cent = (int)c;
+                    break;
+                }
+                case FORMAT_DAY: // %d, %Od - Day of the month, zero-padded (01..31)
+                case FORMAT_DAY_S: // %e, %Oe - Day of the month, blank-padded ( 1..31)
+                {
+                    long d;
+                    if (text.charAt(pos) == ' ') { // brank or not
+                        pos += 1; // brank
+                        d = readDigits(1);
+                    } else {
+                        d = readDigits(2);
+                    }
+
+                    if (!isValidRange(d, 1, 31)) {
+                        values.fail();
+                    }
+                    values.mday = (int)d;
+                    break;
+                }
+                case FORMAT_WEEKYEAR: // %G - The week-based year
+                {
+                    long y;
+                    if (matchAtNumPatterns(nextToken(compiledPattern, i))) {
+                        y = readDigits(4);
+                    } else {
+                        y = readDigitsMax();
+                    }
+                    values.cwyear = (int)y;
+                    break;
```

Is this cast valid? If `readDigitsMax` returns numbers larger than Integer.MAX_VALUE, this cast silently returns broken number.